src/stdlib/RandomMeasures.jl

```diff
@@ -128,41 +128,31 @@ function distribution(d::SizeBiasedSamplingProcess{<:DirichletProcess})
     return LocationScale(zero(α), d.surplus, Beta(one(α), α))
 end
 
-function _logpdf_table(d::DirichletProcess, m::AbstractVector{Int})
+function _logpdf_table(d::DirichletProcess{T}, m::AbstractVector{Int}) where {T<:AbstractFloat}
     # compute the sum of all cluster counts
     sum_m = sum(m)
 
-    # shortcut if all cluster counts are zero
-    dα = d.α
-    T = typeof(dα)
-    iszero(sum_m) && return zeros(T, 1)
-
     # pre-calculations
-    z = log(sum_m - 1 + dα)
+    z = log(sum_m - 1 + d.α)
 
     # construct the table
-    K = length(m)
-    table = Vector{T}(undef, K)
-    contains_zero = false
-    @inbounds for i in 1:K
-        mi = m[i]
-
-        if iszero(mi)
-            if contains_zero
-                table[i] = -Inf
-            else
-                table[i] = log(dα) - z
-                contains_zero = true
-            end
-        else
-            table[i] = log(mi) - z
-        end
+    has_zeros = any(i -> i == 0, m)
```

Comment:
Similar to above, it might be easier to use `iszero`. Moreover, maybe one could add a shortcut here in case `iszero(sum_m)`, i.e.
```suggestion
    has_zeros = (iszero(sum_m) && length(m) > 0) || any(iszero, m)
```
I'm not sure if the case `length(m) == 0` can actually occur/should be handled.

Reply:
See my comment below for a better suggestion, IMO.